### PR TITLE
Events plugin

### PR DESCRIPTION
**Pull request by bkueng**
This brings a plugin for the events protocol (https://mavlink.io/en/services/events.html), allowing to listen for events and arming checks.

- It is using the [component metadata](https://github.com/mavlink/MAVSDK/pull/2272), and therefore moves the implementation to the core, so it's usable for plugins.
- Adds an example for the component metdata & events

### CODE REVIEW

File: examples/component_metadata/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+find_package(MAVSDK REQUIRED)
+
+target_link_libraries(component_metadata
+    MAVSDK::mavsdk
+)
+
+if(NOT MSVC)
+    add_compile_options(component_metadata PRIVATE -Wall -Wextra)
+else()
+    add_compile_options(component_metadata PRIVATE -WX -W2)
```
Comment (julianoes) [inline examples/component_metadata/CMakeLists.txt:21]: No more `-WX` please.

File: examples/events/events.cpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                                  << "]: " << problem.message << std::endl;
+                    }
+                }
+
+                if (verbose && !report.all_problems.empty()) {
+                    std::cout << "All Reports:" << std::endl;
+                    for (const auto& problem : report.all_problems) {
+                        std::cout << "  [" << problem.log_level << "] [" << problem.health_component
+                                  << "]: " << problem.message << std::endl;
+                    }
```
Comment (julianoes) [inline examples/events/events.cpp:107]: Amazing, that's great to have!

File: src/mavsdk/core/curl_test.cpp [hunk 3]
```diff
@@ -82,11 +82,11 @@ TEST_F(CurlTest, Curl_DownloadFile_WithoutProgressFeedback_FileNotFound)
 TEST_F(CurlTest, Curl_DownloadFile_ProgressFeedback_Success)
 {
     int last_progress;
-    Status last_status;
+    HttpStatus last_status;
```
Comment (julianoes) [inline src/mavsdk/core/curl_test.cpp:85]: Whoops, thanks!

File: third_party/libevents/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+foreach(CMAKE_ARG ${CMAKE_ARGS})
+    message(STATUS "-- ${CMAKE_ARG}")
+endforeach()
+
+ExternalProject_Add(
+        libevents
+        GIT_REPOSITORY https://github.com/mavlink/libevents.git
+        GIT_TAG 9474657606d13301d426e044450c4f84de2221be
+        SOURCE_SUBDIR libs/cpp
+        CMAKE_ARGS "${CMAKE_ARGS}"
```
Comment (julianoes) [inline third_party/libevents/CMakeLists.txt:37]: Awesome, that seems pretty clean like that!

File: src/integration_tests/camera_capture_info.cpp [hunk 5]
```diff
@@ -1,5 +1,6 @@
 #include "integration_test_helper.h"
 #include "mavsdk.h"
+#include <algorithm>
```
Comment (julianoes) [inline src/integration_tests/camera_capture_info.cpp:3]: GCC 14 :exploding_head:

**Review by julianoes — COMMENTED** [review]
Amazing, thanks for this! I'll do a more in-depth review of the events internals when CI is ready, and after having reviewed the events API.

File: examples/component_metadata/CMakeLists.txt (hunk 1 above)
Comment (bkueng) [reply examples/component_metadata/CMakeLists.txt]: It's gone

**Comment by bkueng** [comment]
> I'll do a more in-depth review of the events internals when CI is ready, and after having reviewed the events API.

android-x86 had some unrelated request timeout, let's see if it passes now.

**Comment by julianoes** [comment]
Looks like just a fix_style required now...

**Comment by bkueng** [comment]
I think it's because the proto submodule isn't merged/updated yet. At least when I ran it last I didn't get any diff.

**Comment by julianoes** [comment]
Merged proto, rebased, fixed up and force pushed. Let's hope this clears CI.

**Pull request merged**